File: docs/snippets/all/views/dataframe.py
"""Use a blueprint to customize a DataframeView."""

import math

import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_dataframe", spawn=True)

# Log some data.
for t in range(int(math.pi * 4 * 100.0)):
    rr.set_time("t", duration=t)
    rr.log("trig/sin", rr.Scalars(math.sin(float(t) / 100.0)))
    rr.log("trig/cos", rr.Scalars(math.cos(float(t) / 100.0)))

    # some sparse data
    if t % 5 == 0:
        rr.log("trig/tan_sparse", rr.Scalars(math.tan(float(t) / 100.0)))

# Create a Dataframe View
blueprint = rrb.Blueprint(
    rrb.DataframeView(
        origin="/trig",
        query=rrb.archetypes.DataframeQuery(
            timeline="t",
            filter_by_range=(rr.TimeInt(seconds=0), rr.TimeInt(seconds=20)),
            filter_is_not_null="/trig/tan_sparse:Scalar",
            select=["t", "log_tick", "/trig/sin:Scalar", "/trig/cos:Scalar", "/trig/tan_sparse:Scalar"],
            entity_order=["/trig/cos", "/trig/sin", "/trig/tan_sparse"],
            auto_scroll=True,
        ),
    ),
)

rr.send_blueprint(blueprint)

File: docs/snippets/all/views/dataframe.rs
//! Use a blueprint to customize a DataframeView.

use rerun::blueprint::{
    archetypes as blueprint_archetypes, Blueprint, DataframeView,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let blueprint = Blueprint::new(
        DataframeView::new("Dataframe")
            .with_origin("/trig")
            .with_query(
                &blueprint_archetypes::DataframeQuery::new()
                    .with_timeline("t")
                    .with_auto_scroll(true),
            ),
    );

    let rec = rerun::RecordingStreamBuilder::new("rerun_example_dataframe")
        .with_blueprint(blueprint)
        .spawn()?;

    for t in 0..((std::f64::consts::PI * 4.0 * 100.0) as i64) {
        rec.set_duration_secs("t", t as f64);
        rec.log(
            "trig/sin",
            &rerun::Scalars::single((t as f64 / 100.0).sin()),
        )?;
        rec.log(
            "trig/cos",
            &rerun::Scalars::single((t as f64 / 100.0).cos()),
        )?;

        if t % 5 == 0 {
            rec.log(
                "trig/tan_sparse",
                &rerun::Scalars::single((t as f64 / 100.0).tan()),
            )?;
        }
    }

    Ok(())
}
